### PR TITLE
Migrate AccountDeleter job to MSI storage account access

### PR DESCRIPTION
**Pull request by drewgillies**
Addresses: https://github.com/NuGet/Engineering/issues/5586

(Not seeing any reason for the `AccountDeleteConfiguration` delta here in the GitHub view of the diff). Due to complications in the base class DI implementation (see https://github.com/NuGet/Engineering/issues/5606), I've opted for the simple approach of implementing MSI auth via the `UsingMsi` method.

### CODE REVIEW

File: src/AccountDeleter/Configuration/AccountDeleteConfiguration.cs [hunk 1]
```diff
@@ -1,4 +1,4 @@
-﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Copyright (c) .NET Foundation. All rights reserved.
```
Comment (erdembayar) [inline src/AccountDeleter/Configuration/AccountDeleteConfiguration.cs:1]: Looks there's no change in this file. Should we revert the changes?

**Review by erdembayar — APPROVED** [review]

File: src/AccountDeleter/Configuration/AccountDeleteConfiguration.cs (hunk 1 above)
Comment (joelverhagen) [reply src/AccountDeleter/Configuration/AccountDeleteConfiguration.cs:1]: This is a BOM removal, following the editorconfig. I think this change is fine.

**Review by joelverhagen — APPROVED** [review]
